planner/core/find_best_task.go

```diff
@@ -183,6 +183,40 @@ func (p *baseLogicalPlan) findBestTask(prop *property.PhysicalProperty) (bestTas
 	return bestTask, nil
 }
 
+func (p *LogicalUnionScan) findBestTask(prop *property.PhysicalProperty) (bestTask task, err error) {
```

Comment:
Add a method `attach2Task` for `PhysicalUnionScan` is enough?

Reply:
Done.